spring-amqp/src/main/java/org/springframework/amqp/utils/MapBuilder.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.amqp.utils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A {@code Builder} pattern implementation for a {@link Map}.
+ *
+ * @author Artem Bilan
+ * @author Gary Russell
+ * @since 2.0
+ */
+public class MapBuilder<B extends MapBuilder<B, K, V>, K, V> {
```

Comment:
Well, you don't have JavaDocs for generic variable and neither `gredlew api`, nor `gredlew check` barks.
So, in which cases do we have problem with JavaDocs on the matter?